jwst/resample/tests/test_utils.py

```diff
@@ -116,3 +162,43 @@ def test_decode_context():
 
     assert sorted(idx1) == [9, 12, 14, 19, 21, 25, 37, 40, 46, 58, 64, 65, 67, 77]
     assert sorted(idx2) == [9, 20, 29, 36, 47, 49, 64, 69, 70, 79]
+
+
+def test_reproject():
+    gwcs_wcs = create_gwcs()
+    astropy_wcs = fitswcs.WCS(gwcs_wcs.to_fits_sip())
+    xmin, xmax = 100, 500
+    slices = (slice(xmin, xmax), slice(xmin, xmax))
+    sliced_wcs = fitswcs.wcsapi.SlicedLowLevelWCS(gwcs_wcs, slices)
+
+    x = np.arange(150, 200)
+    
+    f = reproject(gwcs_wcs, astropy_wcs)
+    res = f(x, x)
+    assert_allclose(x, res[0], atol=0.1, rtol=0)
+    assert_allclose(x, res[1], atol=0.1, rtol=0)
+
+    f = reproject(astropy_wcs, gwcs_wcs)
+    res = f(x, x)
+    assert_allclose(x, res[0], atol=0.1, rtol=0)
+    assert_allclose(x, res[1], atol=0.1, rtol=0)
+
+    f = reproject(gwcs_wcs, sliced_wcs)
+    res = f(x, x)
+    assert_allclose(x, res[0] + xmin, atol=0.1, rtol=0)
+    assert_allclose(x, res[1] + xmin, atol=0.1, rtol=0)
+
+    f = reproject(sliced_wcs, gwcs_wcs)
+    res = f(x, x)
+    assert_allclose(x, res[0] - xmin, atol=0.1, rtol=0)
+    assert_allclose(x, res[1] - xmin, atol=0.1, rtol=0)
+
+    f = reproject(astropy_wcs, sliced_wcs)
+    res = f(x, x)
+    assert_allclose(x, res[0] + xmin, atol=0.1, rtol=0)
+    assert_allclose(x, res[1] + xmin, atol=0.1, rtol=0)
+
+    f = reproject(sliced_wcs, astropy_wcs)
+    res = f(x, x)
+    assert_allclose(x, res[0] - xmin, atol=0.1, rtol=0)
+    assert_allclose(x, res[1] - xmin, atol=0.1, rtol=0)
```

Comment:
Since all of these have the same pattern, perhaps these could be done with `@pytest.mark.parametrize`?  The 2 wcs objects could be module-scoped fixtures as well.

Reply:
Thanks for the suggestion. I changed those lines to `@pytest.mark.parametrize` to simplify the code for testing.

Reply:
Perfect!